src/Core/Domain/Product/Image/QueryResult/Shop/ShopProductImagesCollection.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Image\QueryResult\Shop;
+
+class ShopProductImagesCollection
+{
+    private $shopProductImages;
+
+    public function __construct(ShopProductImages ...$shopProductImages)
+    {
+        $this->shopProductImages = $shopProductImages;
+    }
+
+    public function getShopProductImagesByShopId(int $shopId): ShopProductImages
```

Comment:
I think we also need a more generic `public function getShopProductImages(): array` because this method is enough for testing in behat since you know in advance which shop you want to check

When we'll display it though we don't so we need to be able to loop through all the shops contained in this collection, so either a `getShopProductImages` for the whole list, or at least we need a `getShopIds` so that we know which shop are contained, and then we can use `getShopProductImagesByShopId` individually

Reply:
Which solution do you prefer?

Reply:
go for `public function getShopProductImages(): array`